stan/math/rev/core/init_chainablestack_tbb.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef STAN_MATH_REV_CORE_INIT_CHAINABLESTACK_TBB_HPP
+#define STAN_MATH_REV_CORE_INIT_CHAINABLESTACK_TBB_HPP
+
+// TODO(SW): remove STAN_THREADS guard once Intel TBB is fully
+// mandatory
+#ifdef STAN_THREADS
+
+#include <stan/math/rev/core/chainablestack.hpp>
+
+#include <tbb/task_scheduler_observer.h>
+
+#include <mutex>
+#include <iostream>
+#include <unordered_map>
+#include <thread>
+#include <utility>
+#include <tuple>
+
+namespace stan {
+namespace math {
+
+static std::mutex cout_mutex;
+
+// initialize AD tape for TBB threads
+struct ad_tape_observer : public tbb::task_scheduler_observer {
+  ad_tape_observer() : tbb::task_scheduler_observer(), ad_tape_() {
+    std::cout << "Observer constructing..." << std::endl;
+    observe(true);  // activate the observer
+    std::cout << "Observer created" << std::endl;
+  }
+
+  ~ad_tape_observer() { std::cout << "Observer destructed" << std::endl; }
+
+  void on_scheduler_entry(bool worker) {
+    std::lock_guard<std::mutex> cout_lock(cout_mutex);
```

Comment:
For clarification, the lock here is for cout and not for anything else here?

Reply:
the lock will go away once the debugging prints are out.